core/vm/contracts.go

```diff
@@ -645,6 +645,8 @@ func (c *getValidator) Run(input []byte, caller common.Address, evm *EVM, gas ui
 
 	index := (&big.Int{}).SetBytes(input[0:32])
 
+	validators := evm.Context.Engine.GetValidators(evm.Context.BlockNumber, common.Hash{})
```

Comment:
What will this return when called during the last block of an epoch? Note that elections are run at the *end* of the block. This may be like `BLOCKHASH` where we can't call it on the current block. Please think carefully about this

Reply:
Semantics are now to retrieve the validator set that needs to sign this block (i.e. we call snapshot with the parent block number and hash).